src/librustc/hir/mod.rs

```diff
@@ -337,6 +337,8 @@ impl fmt::Display for Path {
 pub struct PathSegment {
     /// The identifier portion of this path segment.
     pub ident: Ident,
+    pub id: Option<NodeId>,
+    pub def: Option<Def>,
```

Comment:
Any reason these are `Option`? I'd say fix and/or document them.

Reply:
We don't use them for synthesized path segments. I've added docs.